src/Image.js

```diff
@@ -29,19 +30,28 @@ export default class Image extends React.Component<ImageProps, ImageState> {
 
     static defaultProps = {
         transitionDuration: 300,
-        tint: "dark"
+        tint: "dark",
+        onError: () => {}
     };
 
     state = {
         uri: undefined,
         intensity: new Animated.Value(100)
     };
 
-    async load({uri, options = {}}: ImageProps): Promise<void> {
+    async load({uri, options = {}, onError}: ImageProps): Promise<void> {
         if (uri) {
-            const path = await CacheManager.get(uri, options).getPath();
-            if (this.mounted) {
-                this.setState({ uri: path });
+            try {
+                const path = await CacheManager.get(uri, options).getPath();
```

Comment:
I would consider throwing an Error here: https://github.com/wcandillon/react-native-expo-image-cache/blob/e734867fc8bb386b02ce1e1eafc0d5cbe42288ba/src/CacheManager.js#L33
 in order for your catch to get triggered